webservices/common/models/filings.py

```diff
@@ -123,8 +124,15 @@ def document_description(self):
         )
 
     @property
+    def amended_by(self):
+        if not self.superceded and not self.most_recent and self.amendment and len(self.amendment.longest_chain) > 0:
+            index = self.amendment.longest_chain.index(self.file_number)
+            return self.amendment.longest_chain[index + 1]
```

Comment:
By extract do you mean refactor?  Their purpose is to find the file_number of the filing that amends the current filing being referenced.  The `longest_chain` is the longest amend chain, if it exists, that contains this filing.